NetKet/Operator/local_operator.hpp

```diff
@@ -284,18 +304,34 @@ class LocalOperator : public AbstractOperator {
     sites.insert(sites.end(), rhs.sites_.begin(), rhs.sites_.end());
     mat.insert(mat.end(), rhs.mat_.begin(), rhs.mat_.end());
 
-    return LocalOperator(lhs.GetHilbert(), mat, sites);
+    return LocalOperator(lhs.GetHilbert(), mat, sites,
+                         lhs.constant_ + rhs.constant_);
   }
 
-  void operator=(const LocalOperator &rhs) {
-    assert(rhs.hilbert_.LocalStates().size() == hilbert_.LocalStates().size());
+  friend LocalOperator operator+(const LocalOperator &lhs, double constant) {
+    auto sites = lhs.sites_;
+    auto mat = lhs.mat_;
+
+    return LocalOperator(lhs.GetHilbert(), mat, sites,
+                         lhs.constant_ + constant);
+  }
+
+  LocalOperator &operator+=(const LocalOperator &rhs) {
+    assert(rhs.hilbert_.LocalStates().size() ==
+           this->hilbert_.LocalStates().size());
+
+    this->sites_.insert(this->sites_.end(), rhs.sites_.begin(),
+                        rhs.sites_.end());
+    this->mat_.insert(this->mat_.end(), rhs.mat_.begin(), rhs.mat_.end());
+    this->constant_ += rhs.constant_;
```

Comment:
Is there a reason to add the level shift here? Shouldn't the level shift be consistent for the entire system or is there a situation where you would be adding together two operators with different level shifts?

Reply:
here you add two operators: Osum= O1+O2, where O1=constant1 + sum_i o_i and, O2=constant2 +sum_i o'_i , thus here I am adding the two constants such that constant_sum = constant1+constant2

Reply:
Ok, I'm used to level shifts being applied consistently to all operators, but if we want to allow different level shifts, then this works for me.